packages/form-builder/src/blocks/fields/amount/Edit.tsx

```diff
@@ -1,51 +1,154 @@
+        recurringBillingPeriodOptions,
+        recurringLengthOfTime,
+        recurringOptInDefaultBillingPeriod,
+    } = attributes;
 
     const isMultiLevel = priceOption === 'multi';
     const isFixedAmount = priceOption === 'set';
+    const isRecurringAdmin = recurringEnabled && 'admin' === recurringDonationChoice;
+    const isRecurringDonor = recurringEnabled && 'donor' === recurringDonationChoice;
+    const amountFormatted = formatCurrencyAmount(setPrice);
+
+    const DonationLevels = () => (
+        <LevelGrid>
+            {levels.map((level, index) => {
+                const levelAmount = formatCurrencyAmount(level);
+
+                return <LevelButton key={index}>{levelAmount}</LevelButton>;
+            })}
+        </LevelGrid>
+    );
+
+    const CustomAmount = () => (
+        <CurrencyControl value={setPrice} label={__('Custom amount', 'give')} hideLabelFromVision />
+    );
 
-    const FixedPriceMessage = () => {
-        const amount = formatCurrencyAmount(setPrice);
+    const FixedPriceMessage = () => (
+        <Notice>
+            {createInterpolateElement(__('This donation is set to <amount/> for this form.', 'give'), {
+                amount: <strong>{amountFormatted}</strong>,
+            })}
+        </Notice>
+    );
 
-        return createInterpolateElement(__('This donation is set to <amount/> for this form.', 'give'), {
-            amount: <strong>{amount}</strong>,
-        });
+    const RecurringPeriod = ({count}) => {
+        const interval = count ?? recurringBillingInterval;
+
+        const singular = !isRecurringDonor
+            ? periodLookup[recurringBillingPeriod].singular
+            : periodLookup[recurringOptInDefaultBillingPeriod ?? 'month'].singular;
+
+        const plural = !isRecurringDonor
+            ? periodLookup[recurringBillingPeriod].plural
+            : periodLookup[recurringOptInDefaultBillingPeriod ?? 'month'].plural;
+
+        return (
+            <strong>
+                {1 === interval && <>{singular}</>}
+                {1 !== interval && (
+                    <>
+                        {interval} {plural}
+                    </>
+                )}
+            </strong>
+        );
+    };
+
+    const RecurringMessage = () => {
+
+        const interval = parseInt(recurringBillingInterval);
+
+        const periodPlaceholder = <strong>{'[' + _n('period', 'periods', interval, 'give') + ']'}</strong>;
+
+        const count = parseInt(recurringLengthOfTime);
+        const countElement = <strong>{count}</strong>;
+
+        const translatableString = !count
+            ? __('This donation occurs every <period />.', 'give')
+            : __('This donation occurs every <period /> for <count /> <strong>payments</strong>.', 'give');
+
+        return (
+            <Notice>
+                { 'one-time' === recurringOptInDefaultBillingPeriod
+                    ? __('You have chosen to make this donation one-time.', 'give')
```

Comment:
This is not necessary since it won't translate to the front-end

Reply:
Resolved, see https://github.com/impress-org/givewp-next-gen/pull/128/commits/db3a25cb3973d8e3f56d46409472e4ac3f6a16fd.